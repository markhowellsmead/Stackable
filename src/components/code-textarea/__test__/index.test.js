/**
 * Internal dependencies
 */
import CodeTextarea from '../'

describe( 'Code Textarea', () => {
	test( 'should render correctly', () => {
		expect( shallow( <CodeTextarea /> ) ).toMatchSnapshot()
	} )

	test( 'should display default props', () => {
		const wrapper = shallow( <CodeTextarea /> )
		expect( wrapper.find( 'textarea' ).props().value ).toBe( '' )
	} )

	test( 'should display value', () => {
		const props = {
			value: `abc
line2`,
		}
		const wrapper = shallow( <CodeTextarea { ...props } /> )
		expect( wrapper.find( 'textarea' ).props().value ).toBe( props.value )
	} )

	test( 'should trigger onChange callback', () => {
		const onChange = jest.fn()
		const props = {
			value: 'testing',
			onChange,
		}
		const wrapper = shallow( <CodeTextarea { ...props } /> )
		wrapper.find( 'textarea' ).simulate( 'change', 'New value' )
		expect( onChange ).toHaveBeenCalledWith( 'New value' )
	} )
} )
