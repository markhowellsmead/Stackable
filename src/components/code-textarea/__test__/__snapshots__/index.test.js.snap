// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Textarea should render correctly 1`] = `
<textarea
  id="ugb-code-textarea-1"
  onChange={[Function]}
  value=""
/>
`;
